### PR TITLE
Additional eth methods

### PR DESCRIPTION
**Pull request by F3kilo**
# Issue ticket

Issue ticket link: <>

### Code conventions

- [X] I have performed a self-review of my code
- [X] Every new function is documented
- [X] Object names are auto explicative

### Security

- [X] The PR does not break APIs backward compatibility
- [X] The PR does not break the stable storage backward compatibility

### Testing

- [X] Every function is properly unit tested
- [X] I have added integration tests that prove my fix is effective or that my feature works
- [X] New and existing unit tests pass locally with my changes
- [X] IC endpoints are always tested through the `canister_call!` macro

### CODE REVIEW

File: src/did/src/sign_strategy.rs [hunk 1]
```diff
@@ -95,6 +95,13 @@ impl TransactionSigner for TxSigner {
             Self::ManagementCanister(signer) => signer.sign_transaction(transaction).await,
         }
     }
+
+    async fn sign_digest(&self, digest: [u8; 32]) -> Result<Signature> {
```
Comment (ufoscout) [inline src/did/src/sign_strategy.rs:99]: would it be possible to add a unit test for a Local signer to show the entire loop DIGEST->SIGN->VERIFY

**Review by ufoscout — CHANGES_REQUESTED** [review]

File: src/did/src/sign_strategy.rs [hunk 2]
```diff
@@ -197,7 +211,7 @@ impl TransactionSigner for ManagementCanisterSigner {
         }
 
         let address = IcSigner {}
-            .public_key(self.key_id, DerivationPath::new(vec![]))
+            .public_key(self.key_id, self.derivation_path.clone())
             .await
             .map_err(|e| EvmError::from(format!("failed to get address: {e}")))?;
         let address = H160::from_slice(&address);
```
Comment (ufoscout) [inline src/did/src/sign_strategy.rs:217]: Based on what reported by @Yasir, this line should be replaced with:
```
        let address: H160 = IcSigner
            .pubkey_to_address(&public_key)
            .map_err(|e| {
                EvmError::from(format!(
                    "failed to convert public key to ethereum address: {e}"
                ))
            })?
            .into();
```

**Review by GraDKh — APPROVED** [review]

File: src/did/src/sign_strategy.rs (hunk 2 above)
Comment (F3kilo) [reply src/did/src/sign_strategy.rs]: Done

File: src/did/src/sign_strategy.rs (hunk 1 above)
Comment (F3kilo) [reply src/did/src/sign_strategy.rs:100]: Done

File: src/iceth-client/src/lib.rs [hunk 3]
```diff
@@ -112,7 +129,7 @@ impl Client {
     /// Sends a raw transaction to the external EVM.
     pub async fn send_raw_transaction(&self, transaction_bytes: Bytes) -> Result<H256, Error> {
         let data = format!(
-            r#"{{"jsonrpc":"2.0","id":"3","method":"eth_sendRawTransaction","params":["0x{transaction_bytes:x}"]}}"#,
+            r#"{{"jsonrpc":"2.0","id":4,"method":"eth_sendRawTransaction","params":["0x{transaction_bytes:x}"]}}"#,
```
Comment (ufoscout) [inline src/iceth-client/src/lib.rs:132]: why is the id changed? does it has any particular meaning in this scope?

**Review by ufoscout — APPROVED** [review]

File: src/iceth-client/src/lib.rs (hunk 3 above)
Comment (F3kilo) [reply src/iceth-client/src/lib.rs:132]: No, I've tried to keep id unique for each call.

**Review by itsyaasir — APPROVED** [review]

**Pull request merged**